### ISSUE TITLE
cmd/go: improve error message when working in a module that's outside of go.work

### ISSUE THREAD

**Issue by hyangah** [golang/go#49632]
```
$ go version
go version devel go1.18-3d7cb23e3d Wed Nov 17 01:56:19 2021 +0000 darwin/amd64
```

When running a go command in a directory whose parent directory defines `go.work` but does not include this directory in its use section, the go command output is rather confusing.

```
$ ls
delve
go.work
tools
$ cat go.work
go 1.18

use delve
$ cd tools; go list ./...
pattern ./...: directory prefix . outside available modules
```

Even though users can check `go env GOWORK`, consider to explicitly mention the existence of `go.work`. 
(maybe something like `pattern ./...: directory prefix . outside modules listed in go.work`?)

Or, let it operate in a single module mode as if there is no `go.work` - but not sure which one is less confusing.

**Comment by bcmills** [comment]
CC @matloob

**Comment by ianlancetaylor** [comment]
@matloob This is in the 1.18 milestone; time to move to 1.19?  Thanks.

**Comment by matloob** [comment]
I can fix this for 1.18.

**Comment by perillo** [comment]
What is the reason why the main module is not automatically included in the workspace?
